### ISSUE TITLE
NanoKVM Lite Failed to install runtime library after login

### ISSUE THREAD

**Issue by PetrozPL** [sipeed/NanoKVM#175]
Hi everyone. I have a problem with one of 4 NanoKVMs I've bought. First run with SDCARD is failing after web interface logging in with default admin credentials saying "Failed to install runtime library". There's no prompt to change user and default password as on the other three. Clicking on any links in web interface (terminal, change password etc.) just redirect to login screen, but for terminal it automatically set username to root, but none of credentials working for it. 

![sipeed_error](https://github.com/user-attachments/assets/371e46ba-9d8d-4f77-9564-3efa91b93a54)

Reflashing SD card from scratch does not help. Other three devices that I have worked without any issue, I was able to change the username and password and access terminal. 

Any ideas what is the problem and how to fix it?

**Comment by tzahari** [comment]
Maybe is the SDCARD defect? Try to flash with verification.

**Comment by PetrozPL** [comment]
Hi - I was able to nail down the root cause. It seems that this device requires the internet access during the first power up with newly created SD card. And it seems that you cannot use the SD card initialized on another device, as it will not work. I was initializing those devices online and put them offline as soon as my router were able to detect it and assign IP. But for this one I was using the SD from previous device, and it turned out to be not working, but in the meantime I blacklisted it on my router. Next I flashed SD card from scratch and this time it was not able to finish the initialization in online mode, as it was already blocked on my router. After flashing the card again and unlocking internet access for it I was able to initialize it.

To be honest I have no idea why it needs the internet access for such  thing. Both of those (online initialization requirement and non reusable SD cards from other devices) are suspicious and highly concerning, as any files required for download in online mode could be already put on SD card image.

**Comment by PetrozPL** [comment]
Hi - I was able to nail down the root cause. It seems that this device requires the internet access during the first power up with newly created SD card. And it seems that you cannot use the SD card initialized on another device, as it will not work. I was initializing those devices online and put them offline as soon as my router were able to detect it and assign IP. But for this one I was using the SD from previous device, and it turned out to be not working, but in the meantime I blacklisted it on my router. Next I flashed SD card from scratch and this time it was not able to finish the initialization in online mode, as it was already blocked on my router. After flashing the card again and unlocking internet access for it I was able to initialize it.

To be honest I have no idea why it needs the internet access for such  thing. Both of those (online initialization requirement and non reusable SD cards from other devices) are suspicious and highly concerning, as any files required for download in online mode could be already put on SD card image.
